modules/nf-core/ampcombi/main.nf

```diff
@@ -0,0 +1,45 @@
+process AMPCOMBI {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::ampcombi=0.1.3" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/ampcombi:0.1.3--pyhdfd78af_0':
+        'quay.io/biocontainers/ampcombi:0.1.3--pyhdfd78af_0' }"
+
+    input:
+    tuple val(meta),  path(amp_input)
+    path(faa_folder)
+    val(outdir)
+
+    output:
+    tuple val(meta), path("${outdir}")          , emit: results_dir
+    path("${outdir}/*/*_diamond_matches.txt")   , emit: txt
+    path("${outdir}/*/*_ampcombi.csv")          , emit: csv
+    path("${outdir}/*/*_amp.faa")               , emit: faa
+    path("${outdir}/*.log")                     , optional:true, emit: log
+    path("${outdir}/*")                         , emit: results_subdir
+    path("${outdir}/amp_ref_database")          , optional:true, emit: results_subdir_db
+    path "versions.yml"                         , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def VERSION='0.1.3' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    def fileoption = amp_input.isDirectory() ? "--amp_results $amp_input" : "--path_list $amp_input --sample_list ${prefix}"
```

Comment:
Is it required that the list of sample names comes from prefix? I find this behaviour inconsistent with typical nextflow. I would expect the sample name list should come from a dedicated input channel, and `${prefix}` describes the output file channel name (as this is typically derived from `meta`, which includes an `id` that is a singlular ID for all the given datset).